spyder/plugins/shortcuts/plugin.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2009- Spyder Project Contributors
+#
+# Distributed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+Shortcuts Plugin.
+"""
+
+# Local imports
+from spyder.api.plugins import SpyderPluginV2
+from spyder.api.translations import get_translation
+from spyder.plugins.shortcuts.confpage import ShortcutsConfigPage
+from spyder.plugins.shortcuts.widgets.summary import ShortcutsSummaryDialog
+
+# Localization
+_ = get_translation('spyder')
+
+
+# --- Plugin
+# ----------------------------------------------------------------------------
+class Shortcuts(SpyderPluginV2):
+    """
+    Shortcuts Plugin.
+    """
+
+    NAME = 'shortcuts'
+    # TODO: Fix requires to reflect the desired order in the preferences
+    REQUIRES = []
+    CONTAINER_CLASS = None
+    CONF_WIDGET_CLASS = ShortcutsConfigPage
+    CONF_SECTION = NAME
+    CONF_FILE = False
+
+    # --- SpyderPluginV2 API
+    # ------------------------------------------------------------------------
+    def get_name(self):
+        return _("Keyboard shortcuts")
+
+    def get_description(self):
+        return _("Manage appliaction, widget and actions shortcuts.")
+
+    def get_icon(self):
+        return self.create_icon('keyboard')
+
+    def register(self):
+        pass
+
+    # --- Public API
+    # ------------------------------------------------------------------------
+    def get_shortcut_data(self):
+        """
+        Return the registered shortcut data from the main application window.
+        """
+        return self.main.shortcut_data
+
+    def apply_shortcuts(self):
+        """Apply shortcuts."""
+        self.main.apply_shortcuts()

```

Comment:
This is done right now for convenience, but the idea would be to move methods on the main window to this plugin, so that this plugin is in charge of all the shortcut handling. This goes the same for all the `config` shortcut related stuff.

Reply:
Why can't you that right now?

Reply:
>Why can't you do that right now?

It will require additional changes in the MainWindow that are better suited for another PR.

Reply:
See: https://github.com/spyder-ide/spyder/issues/13527

Reply:
Ok.